exercises/matrix/README.md

```diff
@@ -0,0 +1,87 @@
+# Matrix
+
+Given a string representing a matrix of numbers, return the rows and columns of
+that matrix.
+
+So given a string with embedded newlines like:
```

Comment:
I don't know Fortran, so this might be wrong.

If I understand the test suite correctly, the function takes matrices as input and not a string. I think that's the right way to implement this exercise in Fortran, but the README should be adapted to match the exercise.

Reply:
good point - let me think about this ....

Reply:
I changed the test to parse strings but giving the size of matrix into the function making is somewhat easier not dealing variable dimensions which is also painful in fortran. :-)
When you got the data in fortran, the world is OK, but file-IO is a pain and based on ideas from the 1960s ...

Reply:
hmm, looking at the description it might still not be fully compliant (I am using Array of Strings, not one single string), but I will let you judge.

Reply:
@SaschaMann Have a look at last commit now reading matrices from files. What do you think?

Reply:
Seems reasonable but I think the README should be adjusted.

```
Given a string representing a matrix of numbers, return the rows and columns of
that matrix.

So given a string with embedded newlines like:
```

which doesn't match the fortran implementation of this exercise. I think it could cause some confusion if a student reads the README but then encounters they're given a matrix of numbers represented as characters. I'd suggest changing the README to match the current implementation. The implementation, afaict with my very limited fortran knowledge, seems good. 

Perhaps it could also be a matrix of numbers, rather than a matrix of number-characters, to make it more fitting to Fortran.

Reply:
I changed the implementation once more (this thing is driving me crazy) but I think I am OK with this last state. I use list of strings (character(len=n):: s(m)) and I have added a in the in README.md to reflect this.